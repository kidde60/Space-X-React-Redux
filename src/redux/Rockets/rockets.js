/* eslint-disable */
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

export const fetchRockets = createAsyncThunk(
  'rockets/fetch/rockets',
  async () => {
    const rockets = await fetch('https://api.spacexdata.com/v3/rockets');
    const data = await rockets.json();
    return data;
  },
);

const rocketsSlice = createSlice({
  name: 'rockets',
  initialState: {
    rockets: [],
    status: 'idle',
    error: null,
  },
  reducers: {
    rocketsReserve(state, action) {
      const id = action.payload;
      const ExistingRockets = state.rockets.find((rocket) => rocket.id === id);
      if (ExistingRockets) {
        ExistingRockets.reserved = true;
      }
    },
    rocketsCancel(state, action) {
      const id = action.payload;
      const ExistingRockets = state.rockets.find((rocket) => rocket.id === id);
      if (ExistingRockets) {
        ExistingRockets.reserved = false;
      }
    },
  },
  extraReducers: {
    [fetchRockets.pending]: (state) => {
      state.status = 'loading';
    },
    [fetchRockets.fulfilled]: (state, action) => {
      state.status = 'scceeded';
      state.rockets = [...action.payload];
    },
    [fetchRockets.error]: (state, action) => {
      state.status = 'Error';
      state.error = action.error.message;
    },
  },
});

export default rocketsSlice.reducer;
export const { rocketsReserve, rocketsCancel } = rocketsSlice.actions;
