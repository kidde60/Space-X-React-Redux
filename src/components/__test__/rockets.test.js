import renderer from 'react-test-renderer';
import Rockets from '../rockets/rockets';

const rocket_1 = {
  id: 1,
  rocket_name: "Space X 1",
  describe: "Some Infromation about Space X 1",
  image: "https://imgur.com/DaCfMsj.jpg",
  reserve: true,
  cancel: true,
  reserved: false,
};

it("Check for the first Space X haub", () => {
  const { id, rocket_name, describe, image, reserve, cancel, reserved } =
    rocket_1;
  const firstRocket = renderer
    .create(
      <Rockets
        id={id}
        rocket_name={rocket_name}
        describe={describe}
        image={image}
        reserve={reserve}
        cancel={cancel}
        reserved={reserved}
      />
    )
    .toJSON();
  expect(firstRocket).toMatchSnapshot();
});

const rocket_2 = {
  id: 2,
  rocket_name: 'Space X 2',
  describe: 'More Infromation About Second Space X Hub',
  image: 'https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg',
  reserve: false,
  cancel: true,
  reserved: false,
};
it('Check for second Space X hub', () => {
  const {id, rocket_name, describe, image, reserve, cancel, reserved} = rocket_2
  const secondRocket = renderer.create(<Rockets
    id = {id}
    rocket_name= {rocket_name}
    describe={describe}
    image={image}
    reserve={reserve}
    cancel={cancel}
    reserved ={reserved}
    />).toJSON();
    expect(secondRocket).toMatchSnapshot();
});

const rocket_3 = {
  id: 3,
  rocket_name: 'Space X 3',
  describe: 'More details Aboute space X 3 Hub',
  image: '	https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg',
  reserve: false,
  cancel: true,
  reserved: true
};
it('Check for the Third Space X hub', () => {
  const {id, rocket_name, descrip, image, reserve, cancel, reserved} = rocket_3;
  const thirdRocket = renderer.create(<Rockets 
  id={id}
  rocket_name= {rocket_name}
  describe={descrip}
  image={image}
  reserve ={reserve}
  cancel={cancel}
  reserved ={reserved}
  />)
  expect(thirdRocket).toMatchSnapshot();
});
