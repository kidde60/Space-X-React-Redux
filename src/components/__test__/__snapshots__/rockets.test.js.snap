// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check for second Space X hub 1`] = `
<div
  className="card"
>
  <img
    alt="Space X 2"
    src="https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg"
  />
  <div
    className="rocket_information"
  >
    <h2>
      Space X 2
    </h2>
    <p
      className="reserve-text"
    >
      Not Reserved
    </p>
    <p
      className="rocket-description"
    />
    <div
      className="buttons"
    >
      <button
        className="reserve-btn"
        onClick={[Function]}
        type="button"
      >
        Reserve Racket
      </button>
    </div>
  </div>
</div>
`;

exports[`Check for the Third Space X hub 1`] = `
<div
  className="card"
>
  <img
    alt="Space X 3"
    src="	https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg"
  />
  <div
    className="rocket_information"
  >
    <h2>
      Space X 3
    </h2>
    <p
      className="reserve-text"
    >
      Not Reserved
    </p>
    <p
      className="rocket-description"
    >
      More details Aboute space X 3 Hub
    </p>
    <div
      className="buttons"
    >
      <button
        className="reserve-btn"
        onClick={[Function]}
        type="button"
      >
        Reserve Racket
      </button>
    </div>
  </div>
</div>
`;

exports[`Check for the first Space X haub 1`] = `
<div
  className="card"
>
  <img
    alt="Space X 1"
    src="https://imgur.com/DaCfMsj.jpg"
  />
  <div
    className="rocket_information"
  >
    <h2>
      Space X 1
    </h2>
    <p
      className="reserve-text"
    >
      Not Reserved
    </p>
    <p
      className="rocket-description"
    />
    <div
      className="buttons"
    >
      <button
        className="reserve-btn"
        onClick={[Function]}
        type="button"
      >
        Reserve Racket
      </button>
    </div>
  </div>
</div>
`;
